### ISSUE TITLE
I want to take the frontend part and serve it on a Python (tornado) server. Any idea how I'd start decoupling this from the nodejs server?

### ISSUE THREAD

**Issue by vladiibine** [davezuko/react-redux-starter-kit#1161]
Hi,

First of all, thanks for the repo, it rox. I'm a backend guy, and wanted to try react for a long time, and these starter kits get me up to speed faster.

I'm basically a noob on frontend, so wanted to ask, how I could decouple the frontend part, from the backend one.

I tried to serve the index.html from a simple server, but that doesn't do the trick. I notived something adds extra links in index.html before serving. This is as far as I can tell related to these messages in the console:

```
webpack built 41107f65a2fa9e870671 in 4336ms
Hash: 41107f65a2fa9e870671
Version: webpack 1.14.0
Time: 4336ms
                                Asset       Size  Chunks             Chunk Names
          app.41107f65a2fa9e870671.js     809 kB       0  [emitted]  app
    1.counter.41107f65a2fa9e870671.js    24.9 kB       1  [emitted]  counter
       vendor.41107f65a2fa9e870671.js     390 kB       2  [emitted]  vendor
      app.41107f65a2fa9e870671.js.map     967 kB       0  [emitted]  app
1.counter.41107f65a2fa9e870671.js.map    30.4 kB       1  [emitted]  counter
   vendor.41107f65a2fa9e870671.js.map     461 kB       2  [emitted]  vendor
                          favicon.ico    24.8 kB          [emitted]  
                           index.html  607 bytes          [emitted]  
Child html-webpack-plugin for "index.html":
         Asset    Size  Chunks       Chunk Names
    index.html  578 kB       0       
```

My idea though is that I shouldn't need node.js for running the frontend, so if you have some suggestions of how I'd start to decouple this mess, it would help. Of course, probably I will figure it out on my own, but I thought you might have a kind heart and help a poor developer fellow :)

Thanks!

**Comment by vladiibine** [comment]
Oh, and of course, I don't need instructions regarding the Python/tornado stuff. I'm only new to frontend/ nodejs.

**Comment by jacobpenny** [comment]
```npm run deploy:prod``` will compile the app such that it doesn't depend on the node server and will place the files in the `dist/` directory.

So for example you could do:
```
npm run deploy:prod
cd dist/
python -m SimpleHTTPServer 8001
```
And the app will be available at `localhost:8001`.

**Comment by vladiibine** [comment]
Awesome. Thanks a lot! Ill close the isue tonight when ill test the flow

**Comment by vladiibine** [comment]
Ok, so interestingly enough, the project compiles with hardcoded ports and IP in it. For instance in index.html i see this beauty: "<script type="text/javascript" src="http://192.168.100.6:3000/vendor.41107f65a2fa9e870671.js">".

Not saying i need further assistance, just bringing it to your attention - it's a little unexpected, but probably you can't get rid of that, since I think it's taken from `config/project.config.js`.

Also, I did run `npm run deploy:prod` and it told me to either upgrade node + npm, or there's something wrong in the package
```
0 info it worked if it ends with ok                                            
1 verbose cli [ '/usr/bin/nodejs', '/usr/local/bin/npm', 'run', 'deploy:prod' ]
2 info using npm@4.1.2                                                         
3 info using node@v6.9.5                                                       
4 verbose run-script [ 'predeploy:prod', 'deploy:prod', 'postdeploy:prod' ]    
5 info lifecycle react-redux-starter-kit@3.0.0-alpha.2~predeploy:prod: react-redux-starter-kit@3.0.0-alpha.2
6 silly lifecycle react-redux-starter-kit@3.0.0-alpha.2~predeploy:prod: no script for predeploy:prod, continuing
7 info lifecycle react-redux-starter-kit@3.0.0-alpha.2~deploy:prod: react-redux-starter-kit@3.0.0-alpha.2
8 verbose lifecycle react-redux-starter-kit@3.0.0-alpha.2~deploy:prod: unsafe-perm in lifecycle true
9 verbose lifecycle react-redux-starter-kit@3.0.0-alpha.2~deploy:prod: PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/home/shefuto/pyproj/react-redux-starter-kit/no
de_modules/.bin:/home/shefuto/ve/tmp2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
10 verbose lifecycle react-redux-starter-kit@3.0.0-alpha.2~deploy:prod: CWD: /home/shefuto/pyproj/react-redux-starter-kit
11 silly lifecycle react-redux-starter-kit@3.0.0-alpha.2~deploy:prod: Args: [ '-c', 'better-npm-run deploy:prod' ]
12 silly lifecycle react-redux-starter-kit@3.0.0-alpha.2~deploy:prod: Returned: code: 1  signal: null
13 info lifecycle react-redux-starter-kit@3.0.0-alpha.2~deploy:prod: Failed to exec deploy:prod script
14 verbose stack Error: react-redux-starter-kit@3.0.0-alpha.2 deploy:prod: `better-npm-run deploy:prod`
14 verbose stack Exit status 1                                                 
14 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:279:16)
14 verbose stack     at emitTwo (events.js:106:13)                             
14 verbose stack     at EventEmitter.emit (events.js:191:7)                    
14 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)                             
14 verbose stack     at ChildProcess.emit (events.js:191:7)                    
14 verbose stack     at maybeClose (internal/child_process.js:877:16)          
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid react-redux-starter-kit@3.0.0-alpha.2                         
16 verbose cwd /home/shefuto/pyproj/react-redux-starter-kit                    
17 error Linux 3.13.0-108-generic                                              
18 error argv "/usr/bin/nodejs" "/usr/local/bin/npm" "run" "deploy:prod"       
19 error node v6.9.5                                                           
20 error npm  v4.1.2                                                           
21 error code ELIFECYCLE                                                       
22 error react-redux-starter-kit@3.0.0-alpha.2 deploy:prod: `better-npm-run deploy:prod`
22 error Exit status 1                                                         
23 error Failed at the react-redux-starter-kit@3.0.0-alpha.2 deploy:prod script 'better-npm-run deploy:prod'.
23 error Make sure you have the latest version of node.js and npm installed.   
23 error If you do, this is most likely a problem with the react-redux-starter-kit package,
23 error not with npm itself.                                                  
23 error Tell the author that this fails on your system:                       
23 error     better-npm-run deploy:prod                                        
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs react-redux-starter-kit                                  
23 error Or if that isn't available, you can get their info via:               
23 error     npm owner ls react-redux-starter-kit                              
23 error There is likely additional logging output above.                      
24 verbose exit [ 1, true ]                                                    
```
I'm running on node v6.9.5 and npm 4.1.2...but again, being a noob, I can't tell if I did something wrong, or there's some bug in the project. The thing is `npm start` worked.

Anyway, another thing that didn't work was the counter example. When serving the cp,[iled files from the `dist` folder, the counter page didn't show up, but instead returned 404. I'll check that out myself, but wanted to bring that to you in case you care :D Thanks for the help so far. If you happen ot have any idea of  what I can do quickly to fix these problems, that would be awesome.

**Comment by jacobpenny** [comment]
Not sure why `npm run deploy:prod` would be failing like that, I wasn't able to reproduce.

In the meantime please try
```
NODE_ENV=production npm run compile
```

**Comment by robertzas** [comment]
Try deleting the node_modules folder in the root of your project and do npm cache clean (delete your ~/.npm folder). Then run npm install and npm run deploy:prod. This is the general purpose npm troubleshooting "power-cycle"

**Comment by vladiibine** [comment]
@robertzas I'd say your answer probably worked. I say probably, because while now all my problems are gone, I keep feeling like I get inconsistent behavior. For instance, previously I couldn't get the `/counter` page to work. The browser was doing some kind of request to the server, which didn't have the `./counter` file. Then, after installing another static-file server, namely node's `http-server`, the project worked. But funny enough, it then worked even with the python server. Any idea what could be going on?

**Comment by robertzas** [comment]
@vladiibine What you're probably experiencing is a routing issue. I use this kit as the client with a dot net core kesteral server and have to create a default route that sends everything to the index view except api/*, login/*, etc routes on the actual server (and don't use them on the client). I'm guessing what's happening is you're going to /counter and the server tries to call your server's /counter route, and not the react-router /counter route.

**Comment by vladiibine** [comment]
@robertzas Spot on man! :D I was indeed having the issue that I could navigate to all my links, starting from the home page, but I could never just load the links directly.

Anyway what I did was follow the tutorial for the react-router, and when I got to this lesson (https://github.com/reactjs/react-router-tutorial/tree/master/lessons/11-productionish-server) I realized the exact thing that you're saying: I simply must redirect all the requests to a default route. Anyway, I don't yet exactly grok how this works, but It's part of the complete answer to my question.

Thanks a lot for the help!
